src/core/tracing/trace_context.cpp

```
@@ -0,0 +1,142 @@
+    }
+
+    nixl::trace::TraceContext context;
+    if (!parseBytes(value, 3, context.traceId) || !parseBytes(value, 36, context.spanId) ||
+        !parseByte(value, 53, context.flags) || !context.valid()) {
+        return std::nullopt;
+    }
+    return context;
+}
+
+std::string
+nixl::trace::formatTraceparent(const nixl::trace::TraceContext &context) {
```

Cppcheck (2.21.0)

[style] The function 'formatTraceparent' is never used.

(unusedFunction)